cartoframes/viz/source.py

```diff
@@ -83,48 +76,64 @@ class Source(object):
             Source('table_name', credentials)
     """
 
-    def __init__(self, data, credentials=None, schema=None):
-        if isinstance(data, Dataset):
-            self.dataset = data
-        else:
-            self.dataset = Dataset(data, credentials, schema)
+    def __init__(self, source, credentials=None, schema=None):
```

Comment:
This class inside viz namespace looks weird to me. It is doing a lot of stuff about CARTO and also some viz (basically `compute_metadata`). 

Not sure, but I would split CARTO and viz things in 2 files and moving the CARTO to another place (core?). How do you see it?

Reply:
Yes. Actually, I have created a "SourceManager" class inside "core/managers/source_manager" to manage the query/df duality to be used in the Data Services. The idea is to use also the SourceManager in the Source class.